lib/internal/streams/whatwg.js

```diff
@@ -0,0 +1,4233 @@
+    return readableStreamDefaultControllerCancelSteps(this, reason);
+  }
+
+  [kPull](readRequest) {
+    readableStreamDefaultControllerPullSteps(this, readRequest);
+  }
+
+  [kInspect](depth, options) {
+    return customInspect(
+      depth,
+      options,
+      'ReadableStreamDefaultController', { });
+  }
+
+  get [SymbolToStringTag]() { return 'ReadableStreamDefaultController'; }
+}
+
+class InternalReadableStreamDefaultController {
+  constructor() {
+    this[kState] = {};
+  }
+}
+
+class ReadableByteStreamController {
+  constructor() {
+    throw new ERR_ILLEGAL_CONSTRUCTOR();
+  }
+
+  /**
+   * @readonly
+   * @type {ReadableStreamBYOBRequest}
+   */
+  get byobRequest() {
+    if (this[kState].byobRequest === null &&
+        this[kState].pendingPullIntos.length) {
+      const {
+        buffer,
+        byteOffset,
+        bytesFilled,
+        byteLength,
+      } = this[kState].pendingPullIntos[0];
+      const view =
+        new Uint8Array(
+          buffer,
+          byteOffset + bytesFilled,
+          byteLength - bytesFilled);
+      this[kState].byobRequest =
+        new InternalReadableStreamBYOBRequest(this, view);
+    }
+    return this[kState].byobRequest;
+  }
+
+  /**
+   * @readonly
+   * @type {number}
+   */
+  get desiredSize() {
+    return readableByteStreamControllerGetDesiredSize(this);
+  }
+
+  close() {
+    if (this[kState].closeRequested)
+      throw new ERR_INVALID_STATE.TypeError('Controller is already closed');
+    if (this[kState].stream[kState].state !== 'readable')
+      throw new ERR_INVALID_STATE.TypeError('ReadableStream is already closed');
+    readableByteStreamControllerClose(this);
+  }
+
+  /**
+   * @param {ArrayBufferView} chunk
+   */
+  enqueue(chunk) {
+    if (!isArrayBufferView(chunk)) {
+      throw new ERR_INVALID_ARG_TYPE(
+        'chunk',
+        [
+          'Buffer',
+          'TypedArray',
+          'DataView',
+        ],
+        chunk);
+    }
+    if (chunk.byteLength === 0 || chunk.buffer.byteLength === 0) {
+      throw new ERR_INVALID_ARG_VALUE(
+        'chunk',
+        chunk,
+        'cannot be zero length');
+    }
+    if (this[kState].closeRequested)
+      throw new ERR_INVALID_STATE.TypeError('Controller is already closed');
+    if (this[kState].stream[kState].state !== 'readable')
+      throw new ERR_INVALID_STATE.TypeError('ReadableStream is already closed');
+    readableByteStreamControllerEnqueue(this, chunk);
+  }
+
+  /**
+   * @param {any} error
+   */
+  error(error) {
+    readableByteStreamControllerError(this, error);
+  }
+
+  [kCancel](reason) {
+    return readableByteStreamControllerCancelSteps(this, reason);
+  }
+
+  [kPull](readRequest) {
+    readableByteStreamControllerPullSteps(this, readRequest);
+  }
+
+  [kInspect](depth, options) {
+    return customInspect(
+      depth,
+      options,
+      'ReadableByteStreamController', { });
+  }
+
+  get [SymbolToStringTag]() { return 'ReadableByteStreamController'; }
```

Comment:
why use a non-brand-checking getter here rather than a data property on the prototype?

Reply:
Either works. I was already looking at changing this.